tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ReceiveMessageAsyncIntegration.cs

```diff
@@ -70,7 +75,37 @@ public class ReceiveMessageAsyncIntegration
         /// <param name="state">Calltarget state value</param>
         /// <returns>A response value, in an async scenario will be T of Task of T</returns>
         internal static TResponse OnAsyncMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception exception, in CallTargetState state)
+            where TResponse : IReceiveMessageResponse
         {
+            if (response.Instance != null && response.Messages.Count > 0)
+            {
+                var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+                if (dataStreamsManager != null && dataStreamsManager.IsEnabled)
+                {
+                    var queueName = (string)state.State;
+                    foreach (var o in response.Messages)
+                    {
+                        var message = o.DuckCast<IMessage>();
+                        if (message == null)
+                        {
+                            continue; // should not happen
+                        }
+
+                        var adapter = new ContextPropagation.MessageAttributesAdapter(message.MessageAttributes);
```

Comment:
Definitely want to make this a `readonly struct` if possible as we're creating a new one for every message!